src/vs/workbench/contrib/search/browser/search.contribution.ts

```diff
@@ -62,709 +39,64 @@ registerSingleton(ISearchHistoryService, SearchHistoryService, InstantiationType
-		}
-		return undefined;
-	} else {
-		if (isIncludes) {
-			return commandService.executeCommand(SearchEditorConstants.OpenEditorCommandId, {
-				filesToInclude: (await resolvedResources).join(', '),
-				showIncludesExcludes: true,
-				location: mode === 'newEditor' ? 'new' : 'reuse',
-			});
-		}
-		else {
-			return commandService.executeCommand(SearchEditorConstants.OpenEditorCommandId, {
-				filesToExclude: (await resolvedResources).join(', '),
-				showIncludesExcludes: true,
-				location: mode === 'newEditor' ? 'new' : 'reuse',
-			});
-		}
-	}
-};
-
-const FIND_IN_FOLDER_EXPLORER_ID = 'filesExplorer.findInFolder';
-KeybindingsRegistry.registerCommandAndKeybindingRule({
-	id: FIND_IN_FOLDER_EXPLORER_ID,
-	weight: KeybindingWeight.WorkbenchContrib,
-	when: ContextKeyExpr.and(FilesExplorerFocusCondition, ExplorerFolderContext),
-	primary: KeyMod.Shift | KeyMod.Alt | KeyCode.KeyF,
-	handler: searchInFolderFromExplorer
-});
-
-const FIND_IN_WORKSPACE_ID = 'filesExplorer.findInWorkspace';
-CommandsRegistry.registerCommand({
-	id: FIND_IN_WORKSPACE_ID,
-	handler: async (accessor) => {
-		const searchConfig = accessor.get(IConfigurationService).getValue<ISearchConfiguration>().search;
-		const mode = searchConfig.mode;
-
-		if (mode === 'view') {
-			const searchView = await openSearchView(accessor.get(IViewsService), true);
-			searchView?.searchInFolders();
-		}
-		else {
-			return accessor.get(ICommandService).executeCommand(SearchEditorConstants.OpenEditorCommandId, {
-				location: mode === 'newEditor' ? 'new' : 'reuse',
-				filesToInclude: '',
-			});
-		}
-	}
-});
-
-MenuRegistry.appendMenuItem(MenuId.ExplorerContext, {
-	group: '4_search',
-	order: 10,
-	command: {
-		id: FIND_IN_FOLDER_EXPLORER_ID,
-		title: nls.localize('findInFolder', "Find in Folder...")
-	},
-	when: ContextKeyExpr.and(ExplorerFolderContext)
-});
-
-MenuRegistry.appendMenuItem(MenuId.ExplorerContext, {
-	group: '4_search',
-	order: 10,
-	command: {
-		id: FIND_IN_WORKSPACE_ID,
-		title: nls.localize('findInWorkspace', "Find in Workspace...")
-	},
-	when: ContextKeyExpr.and(ExplorerRootContext, ExplorerFolderContext.toNegated())
-});
-
-class ShowAllSymbolsAction extends Action2 {
-
-	static readonly ID = 'workbench.action.showAllSymbols';
-	static readonly LABEL = nls.localize('showTriggerActions', "Go to Symbol in Workspace...");
-	static readonly ALL_SYMBOLS_PREFIX = '#';
-
-	constructor(
-	) {
-		super({
-			id: Constants.ShowAllSymbolsActionId,
-			title: {
-				value: nls.localize('showTriggerActions', "Go to Symbol in Workspace..."),
-				original: 'Go to Symbol in Workspace...'
-			},
-			f1: true,
-			keybinding: {
-				weight: KeybindingWeight.WorkbenchContrib,
-				primary: KeyMod.CtrlCmd | KeyCode.KeyT
-			}
-		});
-	}
-
-	override async run(accessor: ServicesAccessor): Promise<void> {
-		accessor.get(IQuickInputService).quickAccess.show(ShowAllSymbolsAction.ALL_SYMBOLS_PREFIX);
-	}
-}
-
+// replace/remove actions
+registerAction2(RemoveAction);
```

Comment:
Personally I think it's cleaner to do `registerAction2` and define the class in the same place. The reason is that exporting the action makes it available to anyone else to import, but no other code should ever be importing these actions. However it's something we do differently in pretty much every feature area so it's up to you

Reply:
An example that I like is what we did in notebooks- several files that define actions, organized by category, eg https://github.com/microsoft/vscode/blob/56b7718580777e16b8305dbe9d50b9773dc759d9/src/vs/workbench/contrib/notebook/browser/controller/layoutActions.ts

Reply:
changed to call `registerAction2` in multiple files that get imported into `search.contributions.ts`